src/java.base/share/classes/jdk/internal/event/SocketReadEvent.java

```diff
@@ -0,0 +1,131 @@
+    // The order must match the argument order of the generated commit method.
+    public String host;
+    public String address;
+    public int port;
+    public long timeout;
+    public long bytesRead;
+    public boolean endOfStream;
+
+    /**
+     * Actually commit a socket read event.  The implementation
+     * of this method is generated automatically if jfr is enabled.
+     * The order of the fields must be the same as the parameters in this method.
+     * {@code commit(..., String, String, int, long, long, boolean)}
+     *
+     * @param start     timestamp of the start of the operation
+     * @param duration  time in nanoseconds to complete the operation
+     * @param host      remote host of the transfer
+     * @param address   remote address of the transfer
+     * @param port      remote port of the transfer
+     * @param timeout   timeout setting for the read
+     * @param bytes     number of bytes that were transferred
+     * @param endOfStream  has the end of the stream been reached
+     */
+    public static void commit(long start, long duration, String host, String address, int port, long timeout, long bytes, boolean endOfStream) {
+        // Generated by JFR
+    }
+
+    /**
+     * Determine if an event should be emitted.  The duration of the operation
+     * must exceed some threshold in order to commit the event.  The implementation
+     * of this method is generated automatically if jfr is enabled.
+     *
+     * @param duration  time in nanoseconds to complete the operation
+     * @return  true if the event should be commited
+     */
+    public static boolean shouldCommit(long duration) {
+        // Generated by JFR
+        return false;
+    }
+
+    /**
+     * Determine if this kind of event is enabled.  The implementation
+     * of this method is generated automatically if jfr is enabled.
+     *
+     * @return true if socket read events are enabled, false otherwise
+     */
+    public static boolean enabled() {
+        // Generated by JFR
+        return false;
+    }
+
+    /**
+     * Fetch the current timestamp in nanoseconds.  This method is used
+     * to determine the start and end of an operation.  The implementation
+     * of this method is generated automatically if jfr is enabled.
+     *
+     * @return  the current timestamp value
+     */
+    public static long timestamp() {
+        // Generated by JFR
+        return 0L;
+    }
+
+    /**
+     * Execute the standard boilerplate that proceeds a potential call to the machine generated
+     * commit method.
+     *
+     * @param start  the start time
+     * @param nbytes  how many bytes were transferred
+     * @param remote  the address of the remote socket
+     * @param timeout  maximum time to wait
+     */
+    public static void checkForCommit(long start, long nbytes, SocketAddress remote, long timeout) {
+        long duration = timestamp() - start;
+        if (shouldCommit(duration)) {
```

Comment:
I think you know this, but this will need to be re-examined for SocketChannel configured non-blocking.